### PR TITLE
TravisCI: Automatically update "nodegit.github.com" repository

### PR DESCRIPTION
**Pull request by Turbo87**
as suggested in #787

To make this work you will have to generate a GitHub API token with the `public_repo` permission and add it as a secret environment variable `GH_TOKEN` to the TravisCI project settings. After that every push to the `master` branch should automatically update the https://github.com/nodegit/nodegit.github.com repository.

### CODE REVIEW

**Comment by Turbo87** [comment]
Since I only changed the Linux + Node.js 4.1 instructions the build failure on TravisCI is most likely not related to my changes. If you want to make sure everything works as expected please restart the failed jobs to try again.

**Comment by tbranyen** [comment]
Can you add a tag with the value of TRAVIS_COMMIT? I'd like to wire up a nav in the website to list all the tags (avoids listing all commits) and it'd be cool to search by NodeGit SHA.

**Comment by Turbo87** [comment]
@tbranyen sorry, could you rephrase that? I'm not sure what you mean.

**Comment by Turbo87** [comment]
I rebased the branch to get rid of the merge conflict created by #792

**Comment by Turbo87** [comment]
@tbranyen I think I understand what you want. Should be possible, although I'm not sure if it will be all that useful. I think it would make more sense to have have the same tags in the website repo as in the main repo.

**Comment by tbranyen** [comment]
```
git tag $TRAVIS_VERSION
git push $TRAVIS_VERSION
```

Is what I'm thinking.

**Comment by Turbo87** [comment]
@tbranyen what do you mean by `$TRAVIS_VERSION`? (not mentioned in http://docs.travis-ci.com/user/environment-variables/)

Again: do you really want to tag every single commit in the repo? or do you want the tags in the website repo to be similar to the tags in the main repo?

**Comment by tbranyen** [comment]
@Turbo87 sorry, was a typo when I writing on the road. I updated my comment to show `TRAVIS_COMMIT`

**Comment by Turbo87** [comment]
@tbranyen done

**Comment by tbranyen** [comment]
lgtm :+1:

**Comment by tbranyen** [comment]
Thanks a ton!

**Comment by Turbo87** [comment]
hmm, it doesn't seem to upload anything yet though... did you add the `GH_TOKEN` env var yet?

**Comment by tbranyen** [comment]
No not yet, I didn't realize that was a requirement for us here. Shouldn't that just be an encrypted Travis value in the .travis.yml?

**Comment by johnhaley81** [comment]
Yeah, I'd rather encrypt the value and put it in `.travis.yml` than have it live on the Travis web site.

**Comment by Turbo87** [comment]
@tbranyen feel free to do that. I'm not sure if I'm able to encrypt it myself without write access to the repo.
